tests/contest/contest/src/tests/linux_masked_paths/masked_paths.rs

```diff
@@ -0,0 +1,251 @@
+    let masked_file_top = PathBuf::from(masked_file);
+
+    let masked_dir_sub = masked_dir_top.join(masked_subdir);
+    let masked_file_sub = masked_dir_top.join(masked_file);
+    let masked_file_sub_sub = masked_dir_sub.join(masked_file);
+
+    let root = PathBuf::from("/");
+
+    let masked_paths = vec![
+        root.join(&masked_dir_top).to_string_lossy().to_string(),
+        root.join(&masked_file_top).to_string_lossy().to_string(),
+        root.join(&masked_dir_sub).to_string_lossy().to_string(),
+        root.join(&masked_file_sub).to_string_lossy().to_string(),
+        root.join(&masked_file_sub_sub)
+            .to_string_lossy()
+            .to_string(),
+    ];
+
+    let spec = get_spec(masked_paths);
+
+    test_inside_container(spec, &|bundle_path| {
+        use std::fs;
+        let test_dir = bundle_path.join(&masked_dir_sub);
+        if let Err(e) = fs::create_dir_all(&test_dir) {
+            bail!(e)
+        }
+
+        if let Err(e) = fs::File::create(test_dir.join("tmp")) {
+            bail!(e)
+        }
+
+        // runtimetest cannot check the readability of empty files, so
+        // write something.
+        let test_sub_sub_file = bundle_path.join(&masked_file_sub_sub);
+        if let Err(e) = fs::File::create(&test_sub_sub_file) {
+            bail!(e)
+        }
+        if let Err(e) = fs::write(&test_sub_sub_file, b"secrets") {
+            bail!(e)
+        }
+
+        let test_sub_file = bundle_path.join(&masked_file_sub);
+        if let Err(e) = fs::File::create(&test_sub_file) {
+            bail!(e)
+        }
+        if let Err(e) = fs::write(&test_sub_file, b"secrets") {
+            bail!(e)
+        }
+
+        let test_file = bundle_path.join(masked_file);
+        if let Err(e) = fs::File::create(&test_file) {
+            bail!(e)
+        }
+        if let Err(e) = fs::write(&test_file, b"secrets") {
+            bail!(e)
+        }
+
+        Ok(())
+    })
+}
+
+fn check_masked_rel_paths() -> TestResult {
+    // Deliberately set a relative path to be masked,
+    // and expect an error
+    let masked_rel_path = "masked_rel_path";
+    let masked_paths = vec![masked_rel_path.to_string()];
+    let spec = get_spec(masked_paths);
+
+    test_inside_container(spec, &|bundle_path| {
+        use std::{fs, io};
+        let test_file = bundle_path.join(masked_rel_path);
+        match fs::metadata(&test_file) {
+            io::Result::Ok(md) => {
+                bail!(
+                    "reading path {:?} should have given error, found {:?} instead",
+                    test_file,
+                    md
+                )
+            }
+            io::Result::Err(e) => {
+                let err = e.kind();
+                if let io::ErrorKind::NotFound = err {
+                    Ok(())
+                } else {
+                    bail!("expected not found error, got {:?}", err);
+                }
+            }
+        }
+    })
```

Comment:
Why is this check here? we should be checking this in the runtime test and here we should be creating the path, right? Or did I misunderstood anything?

Reply:
When setting a masked path, it is impossible to determine whether it is a relative path, so I added this to verify whether it meets the test requirements at the time of creation. This implementation is also adopted in runtime-tools, so verification is necessary. However, I overlooked a few points regarding file creation, so I will make corrections.

Reply:
ok, so I see that you have carried over the implementation from go tests, but I'm still having a bit of difficulty understanding this -
1. The go code seems to checks  that if error is not null and error is not found, then return error, else return null. I.e. if the error is not found, then the prepare function will return error, other wise not. Here the behavior seems to be opposite.
2. Also, this function/closure we pass, both here and in go code, will be executed before runtime has done anything. This is supposed to prepare the rootfs for tests running inside the container. Given that, what exactly is this checking? This will be running on plain bundle fs that is created for each test, It is ofcourse not going to find any of the files. 

What I'm thinking here is that even the go tests have slightly incorrect implementation (which has happened before). Can you confirm going through comments in go code and seeing what the test is supposed to check, and what it is actually validating?

Reply:
There are cases where the Go implementation may be incorrect. I'll check it out.

Reply:
I posted a question in the Issue because I couldn’t fully grasp the purpose of this test.（https://github.com/opencontainers/runtime-tools/issues/780 ）

This is my own opinion, but I believe the test might be trying to illustrate a discrepancy where the path added to `masked_path` is specified from the host system’s perspective, resulting in an unintended relative path from `/` inside the container that doesn’t match the expected path.

As for addressing this test, we could:
1. Add the above comment
2. Wait for a response on the Issue before deciding
3. Delete the test
4. Refactor the test

If we choose to improve the test, I think the following approach could work:
- Create a file inside the container we want to mask (`masked-relpath`).
- Specify the relative path from the host system (`./bundle/rootfs/masked-relpath`) as the path to mask.
- After creating the container, verify that the file is masked by checking if it exists at the expected path without the `./bundle/rootfs` prefix.

Do you think this would be a good approach?

Reply:
As stated in the [runtime-spec](https://github.com/opencontainers/runtime-spec/blob/main/config-linux.md#masked-paths), setting a relative path is not allowed. Therefore, I believe the correct responsibility of contest is to produce an error when a relative_path is set during testing. Based on this, I implemented the following.
d1fa11b

Regarding `runtimetest`, I made it verify that paths passed to the namespace are not relative path to ensure the behavior complies with the [runtime-spec](https://github.com/opencontainers/runtime-spec/blob/main/config-linux.md#masked-paths).

One concern is whether to add an implementation to validate relative_path in the spec's validation. This would also affect the test behavior, so I am leaving this comment for discussion.
https://github.com/youki-dev/youki/blob/main/crates/libcontainer/src/container/init_builder.rs#L152

BTW, in `runc`, validation to ensure `masked_path` is an absolute path is not performed, so it might not be necessary.